lib/mix/tasks/docs.ex

```diff
@@ -113,25 +113,35 @@ defmodule Mix.Tasks.Docs do
   def run(args, config \\ Mix.Project.config, generator \\ &ExDoc.generate_docs/3) do
     Mix.Task.run "compile"
 
-    {cli_opts, args, _} = OptionParser.parse(args,
-                            aliases: [n: :canonical, f: :formatter, o: :output],
-                            switches: [canonical: :string, formatter: :keep, language: :string, output: :string])
-
-    if args != [] do
-      Mix.raise "Extraneous arguments on the command line"
+    {cli_options, remaining_args, _invalid} =
```

Comment:
This section does not seems to be related with your original goal, which is add `ExDoc.normalize_options/3`.